change Ubuntu to Debian-based distros (#4376)
rust-lang/book: src/ch01-01-installation.md

```

Linux users should generally install GCC or Clang, according to their

> distribution’s documentation. For example, if you use Debian-based distros, you can install
> the `build-essential` package.

Reviewer: If we're adjusting to a more generic example, might as well provide the install command as well:

````suggestion
distribution’s documentation. For example, in Debian-based distros you can install
the `build-essential` package via:

```console
$ sudo apt install build-essential
```
````

Author: I agree, but omitted "via" because I believe it helps with clarity (keeps the connection with the previous sentence better).
